scripts/loop.py

```
@@ -119,7 +113,17 @@ def main(argv: list[str]) -> int:
     ]
     if owner_prompt:
         cmd.extend(["--append-system-prompt", owner_prompt])
-    subprocess.run(cmd, check=False)
+    # `subprocess.run(timeout=N)` portably terminates the child after N seconds
+    # on every supported OS, replacing the external GNU `timeout(1)` command
+    # which is missing on default macOS and is a Windows shell built-in with
+    # incompatible semantics.
+    try:
+        subprocess.run(cmd, check=False, timeout=timeout)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)